--- terminal-screen.c.orig	2016-10-26 07:28:37.034611917 -0700
+++ terminal-screen.c	2016-10-26 07:31:18.237486372 -0700
@@ -146,6 +146,8 @@
 static gboolean terminal_screen_popup_menu (GtkWidget *widget);
 static gboolean terminal_screen_button_press (GtkWidget *widget,
                                               GdkEventButton *event);
+static gboolean terminal_screen_button_release (GtkWidget *widget,
+                                                GdkEventButton *event);
 static gboolean terminal_screen_do_exec (TerminalScreen *screen,
                                          FDSetupData    *data,
                                          GError **error);
@@ -600,6 +602,7 @@
   widget_class->style_updated = terminal_screen_style_updated;
   widget_class->drag_data_received = terminal_screen_drag_data_received;
   widget_class->button_press_event = terminal_screen_button_press;
+  widget_class->button_release_event = terminal_screen_button_release;
   widget_class->popup_menu = terminal_screen_popup_menu;
 
   terminal_class->child_exited = terminal_screen_child_exited;
@@ -1800,9 +1803,7 @@
           if (button_press_event && button_press_event (widget, event))
             return TRUE;
 
-          terminal_screen_do_popup (screen, event, url, url_flavor, number_info);
-          url = NULL; /* adopted to the popup info */
-          number_info = NULL; /* detto */
+          vte_terminal_paste_clipboard (VTE_TERMINAL (screen));
           return TRUE;
         }
       else if (!(event->state & (GDK_CONTROL_MASK | GDK_MOD1_MASK)))
@@ -1822,6 +1823,33 @@
   return FALSE;
 }
 
+static gboolean
+terminal_screen_button_release (GtkWidget      *widget,
+                                GdkEventButton *event)
+{
+    gboolean ret;
+
+    TerminalScreen *screen = TERMINAL_SCREEN (widget);
+    gboolean (* button_release_event) (GtkWidget*, GdkEventButton*) =
+        GTK_WIDGET_CLASS (terminal_screen_parent_class)->button_release_event;
+
+    ret = FALSE;
+    if (button_release_event) {
+        ret = button_release_event (widget, event);
+    }
+
+    if (event->button == 1) {
+        gboolean can_copy;
+
+        can_copy = vte_terminal_get_has_selection (VTE_TERMINAL (screen));
+
+        if (can_copy)
+            vte_terminal_copy_clipboard (VTE_TERMINAL (screen));
+    }
+
+    return ret;
+}
+
 /**
  * terminal_screen_get_current_dir:
  * @screen:
