Description: This gnome-terminal supports paste by right-click

Index: gnome-terminal/src/terminal-screen.c
===================================================================
--- terminal-screen-orig.c	2015-11-08 16:31:54.357079394 -0800
+++ terminal-screen.c	2015-11-08 16:34:32.523900858 -0800
@@ -135,6 +135,8 @@
 static gboolean terminal_screen_popup_menu (GtkWidget *widget);
 static gboolean terminal_screen_button_press (GtkWidget *widget,
                                               GdkEventButton *event);
+static gboolean terminal_screen_button_release (GtkWidget *widget,
+                                                GdkEventButton *event);
 static gboolean terminal_screen_do_exec (TerminalScreen *screen,
                                          FDSetupData    *data,
                                          GError **error);
@@ -521,6 +523,7 @@
   widget_class->realize = terminal_screen_realize;
   widget_class->drag_data_received = terminal_screen_drag_data_received;
   widget_class->button_press_event = terminal_screen_button_press;
+  widget_class->button_release_event = terminal_screen_button_release;
   widget_class->popup_menu = terminal_screen_popup_menu;

   terminal_class->child_exited = terminal_screen_child_exited;
@@ -1654,8 +1657,7 @@
           if (button_press_event && button_press_event (widget, event))
             return TRUE;

-          terminal_screen_do_popup (screen, event, matched_string, matched_flavor);
-          matched_string = NULL; /* adopted to the popup info */
+          vte_terminal_paste_clipboard (VTE_TERMINAL (screen));
           return TRUE;
         }
       else if (!(event->state & (GDK_CONTROL_MASK | GDK_MOD1_MASK)))
@@ -1674,6 +1676,33 @@
   return FALSE;
 }

+static gboolean
+terminal_screen_button_release (GtkWidget      *widget,
+                                GdkEventButton *event)
+{
+    gboolean ret;
+
+    TerminalScreen *screen = TERMINAL_SCREEN (widget);
+    gboolean (* button_release_event) (GtkWidget*, GdkEventButton*) =
+        GTK_WIDGET_CLASS (terminal_screen_parent_class)->button_release_event;
+
+    ret = FALSE;
+    if (button_release_event) {
+        ret = button_release_event (widget, event);
+    }
+
+    if (event->button == 1) {
+        gboolean can_copy;
+
+        can_copy = vte_terminal_get_has_selection (VTE_TERMINAL (screen));
+
+        if (can_copy)
+            vte_terminal_copy_clipboard (VTE_TERMINAL (screen));
+    }
+
+    return ret;
+}
+
 /**
  * terminal_screen_get_current_dir:
  * @screen:

Index: gnome-terminal/README
===================================================================
--- /dev/null
+++ gnome-terminal/README
@@ -0,0 +1 @@
+This gnome-terminal supports paste by right-click
